src/track/trackinfo.cpp

```diff
@@ -133,4 +134,34 @@ QDebug operator<<(QDebug dbg, const TrackInfo& arg) {
     return dbg;
 }
 
+// Returns the list of genres for this track.
+const QList<Genre>& TrackInfo::getGenres() const {
+    return m_genres;
+}
+
+// Returns a comma-separated string of genre names for simple display.
+QString TrackInfo::getGenresString() const {
+    QStringList genreNames;
+    // Reserve memory to optimize performance for tracks with many genres.
+    genreNames.reserve(m_genres.size());
+    for (const Genre& genre : m_genres) {
+        genreNames.append(genre.name);
+    }
+    // Joins the names with ", " for a readable format, e.g. "House, Techno".
+    return genreNames.join(", ");
+}
+
+// Sets the list of genres for this track.
+void TrackInfo::setGenres(const QList<Genre>& genres) {
+    if (m_genres != genres) {
+        m_genres = genres;
+    }
+}
+
+// Debug helper to print genre info, replacing the auto-generated dbgGenre.
+void TrackInfo::dbgGenres(QDebug& dbg) const {
+    dbg.nospace() << "," << Qt::endl
+                  << "  genres: " << getGenresString();
+}
```

Comment:
You probably want to an operator overload: 

QDebug operator<<(QDebug debug,  QList<Genre>& genres);

like here: 

https://github.com/mixxxdj/mixxx/blob/6a834ceedcff41527df59651cfccc6712f5ed9b3/src/track/cueinfo.cpp#L155